src/hooks/useFetchListCallback.ts

```diff
@@ -6,26 +6,21 @@ import resolveENSContentHash from 'lib/utils/resolveENSContentHash'
 import { useCallback } from 'react'
 import { useAppDispatch } from 'state/hooks'
 
-import { getNetworkLibrary } from '../connectors'
 import { fetchTokenList } from '../state/lists/actions'
 
 export function useFetchListCallback(): (listUrl: string, sendDispatch?: boolean) => Promise<TokenList> {
-  const { chainId, library } = useActiveWeb3React()
+  const { provider } = useActiveWeb3React()
   const dispatch = useAppDispatch()
 
   const ensResolver = useCallback(
     async (ensName: string) => {
-      if (!library || chainId !== 1) {
-        const networkLibrary = getNetworkLibrary()
-        const network = await networkLibrary.getNetwork()
-        if (networkLibrary && network.chainId === 1) {
-          return resolveENSContentHash(ensName, networkLibrary)
-        }
-        throw new Error('Could not construct mainnet ENS resolver')
+      if (provider) {
+        return resolveENSContentHash(ensName, provider)
+      } else {
+        return Promise.resolve('')
```

Comment:
Could you have a separate mainnet connector (or provider) to do this? It doesn't necessarily need to go through @web3-react - it _could_ just be an ethers url provider for mainnet.

Reply:
good point, this is doable